src/DateRange.js

```diff
@@ -8,43 +8,203 @@ const defaultOptions = [
         this.setState({
-            startDate: selection.value
+            startDate: selection
         });
-        this.props.onDateRangeChanged(this.state.startDate, this.state.endDate);
+        this.props.onDateRangeChanged(selection.value, this.state.endDate.value);
     }
 
     onMaxDateChanged(selection)
     {
         this.setState({
-            endDate: selection.value
+            endDate: selection
+        });
+        this.props.onDateRangeChanged(this.state.startDate.value, selection.value);
+    }
+
+    setLocation(location)
+    {
+        this.setState({
+            location: {
+                "category":  location.category,
+                "name": location.name
+            }
         });
-        this.props.onDateRangeChanged(this.state.startDate, this.state.endDate);
+        this.updateValidDateRange(location.category, location.name);
+    }
+
+    updateValidDateRange(locationCategory, locationName)
+    {
+        console.log("updatevaliddaterange query for: ", locationCategory, locationName);
+        if (!locationCategory && !locationName){
+            return;
+        }
+
+        let url = `http://coa-flask-app-dev.us-east-1.elasticbeanstalk.com`
+        let tail = `/validdaterange`
+            + `?locationCategory=` + locationCategory
+            + `&locationName=` + locationName
+
+        let responseHandler =
+            function(results) {
+                results.json().then(this.handleValidDateRangeResponse.bind(this));
+            }.bind(this);
+
+        fetch(url + tail,
+            {"method": 'GET', "mode": "cors"}) 
+        .then(responseHandler)
+        .catch(
+            function() {
+                console.log("Failed to hit deployed service for dirty dozen api, trying to hit the api locally.");
+                fetch(`http://127.0.0.1:5000` + tail, {"method": 'GET', "mode": "cors"})
+                .then(responseHandler)
+                .catch(function(){
+                    console.log("Failed to execute query for valid date range.")
+                });
+            }
+        );
+    }
+
+    getStartDateOption(season, year)
+    {
+        return {
+            "label": season.label + ' ' + year.toString(),
+            "value": year.toString() + '-' + season.startDay
+        };
+    }
+
+    getEndDateOption(season, year)
+    {
+        let endYear = (season.label === 'Winter') ? year + 1 : year;
+        return {
+            "label": season.label + ' ' + year.toString(),
+            "value": endYear.toString() + '-' + season.endDay
+        };
+    }
+
+    handleValidDateRangeResponse(data)
+    {
+        console.log("validdaterange response", data.validDateRange.firstDate, data.validDateRange.lastDate);
+        let firstDate = data.validDateRange.firstDate;
+        let lastDate = data.validDateRange.lastDate;
+
+        let seasonIdx = 0;
+        let season = SEASON_ORDER[seasonIdx];
+        let year = parseInt(firstDate.substr(0, 4));
```

Comment:
Its best practice to specify the radix when using `parseInt` (i.e. `parseInt(date, 10)`. Default radix is implementation dependent.